### PR TITLE
Update HDFS: Support for erasure codes

### PR DESCRIPTION
**Pull request by M1eyu2018**
**Changelog category (leave one):**

- Improvement

**Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):**
Support for erasure codes in HDFS, author: @M1eyu2018, @tomscut.

### CODE REVIEW

**Comment by M1eyu2018** [comment]
https://github.com/ClickHouse/ClickHouse/pull/48773

File: contrib/libhdfs3-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -148,6 +172,12 @@ if (TARGET OpenSSL::SSL)
     target_link_libraries(_hdfs3 PRIVATE OpenSSL::Crypto OpenSSL::SSL)
 endif()
 
+find_library(ISAL_LIBRARY NAMES isal)
```
Comment (alexey-milovidov) [inline contrib/libhdfs3-cmake/CMakeLists.txt:175]: We don't allow `find_library`
All libraries should be imported to `contrib`

**Comment by M1eyu2018** [comment]
hi, @alexey-milovidov. I have fixed the failed tests in libhdfs3 https://github.com/ClickHouse/libhdfs3/pull/44, please review it again.
And now there is no failed test about hdfs.

**Comment by alexey-milovidov** [comment]
Ok, let's update this PR.

**Comment by M1eyu2018** [comment]
> Ok, let's update this PR.

@alexey-milovidov all checks have been run. Please review the pr. By the way, could you fork intel-isal into clickhouse as submodule?

**Comment by alexey-milovidov** [comment]
Ok. It does not build with sanitizers, because the `-fsanitize` option is passed to `nasm` while it shouldn't.

**Comment by alexey-milovidov** [comment]
https://github.com/ClickHouse/isa-l

**Comment by M1eyu2018** [comment]
> Ok. It does not build with sanitizers, because the `-fsanitize` option is passed to `nasm` while it shouldn't.

@alexey-milovidov, I don't understand it. What can I do for it?
remove options -fno-sanitize=undefined in CmakeLists like contrib/qpl-cmake/CMakeLists.txt?
企业微信截图_bf47b251-6044-472f-a9e3-5cc2cb7823f6

**Comment by M1eyu2018** [comment]
> > Ok. It does not build with sanitizers, because the `-fsanitize` option is passed to `nasm` while it shouldn't.
> 
> @alexey-milovidov, I don't understand it. What can I do for it? Remove options -fno-sanitize=undefined in CmakeLists like contrib/qpl-cmake/CMakeLists.txt? 企业微信截图_bf47b251-6044-472f-a9e3-5cc2cb7823f6

**Comment by alexey-milovidov** [comment]
@M1eyu2018, we need to somehow don't pass these arguments to nasm.

**Comment by taiyang-li** [comment]
@M1eyu2018 I think it is better to add other collegues of your group to author list.

**Pull request merged**